process/peer/process.go

```diff
@@ -325,10 +331,33 @@ func (vs *validatorStatistics) checkForMissedBlocks(
 	prevRandSeed []byte,
 	shardId uint32,
 ) error {
-	if currentHeaderRound-previousHeaderRound <= 1 {
+	missedRounds := currentHeaderRound - previousHeaderRound
```

Comment:
you can refactor this method to reduce the code complexity


	tooManyComputations := missedRounds > vs.maxComputableRounds
	if !tooManyComputations {
		return vs.computeDecrease(previousHeaderRound, currentHeaderRound, prevRandSeed, shardId)
	}

	consensusGroup, err := vs.nodesCoordinator.ComputeValidatorsGroup(prevRandSeed, previousHeaderRound+1, shardId)
	if err != nil {
		return err
	}

	consensusGroupSize := len(consensusGroup)
	return vs.decreaseAll(shardId, missedRounds, consensusGroupSize)`

Reply:
:+1:

Reply:
fixed